### PR TITLE
Resolve conflicts in files for test cases 

### PR DESCRIPTION
**Pull request by simrin051**
It has test files ( spec files ) and certain changes had to be made to html file and ts files to pass test cases. All the needed files are committed in this PR.

### CODE REVIEW

**Comment by simrin051** [comment]
@arey 

Can you please check now. Please.

File: src/app/owners/owner.service.spec.ts [hunk 1]
```diff
@@ -87,33 +95,117 @@ describe('OnwerService', () => {
     httpTestingController.verify();
   });
 
-  /// OwnerService method tests begin ///
-
-  describe('#getOwners', () => {
-    let expectedOwners: Owner[];
+  it("should return expected owners (called once)", () => {
```
Comment (arey) [inline src/app/owners/owner.service.spec.ts:98]: Did you have use the linter (`ng lint`) ? I remember single quote should be used.

File: wq [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[33mea40ba4[m[33m ([m[1;36mHEAD -> [m[1;32mtestcases[m[33m, [m[1;31morigin/testcases[m[33m)[m HEAD@{0}: reset: moving to ea40ba4
```
Comment (arey) [inline wq:1]: Could you remove this file ?

File: src/app/specialties/specialty.service.ts [hunk 3]
```diff
@@ -20,58 +20,55 @@
  * @author Vitaliy Fedoriv
```
Comment (arey) [inline src/app/specialties/specialty.service.ts:20]: Could you revert reformating please ?

File: src/app/specialties/specialty-add/specialty-add.component.ts [hunk 4]
```diff
@@ -16,39 +16,47 @@
  *
  */
 
-import {Component, EventEmitter, OnInit, Output} from '@angular/core';
-import {Specialty} from '../specialty';
-import {SpecialtyService} from '../specialty.service';
+import {
```
Comment (arey) [inline src/app/specialties/specialty-add/specialty-add.component.ts:19]: Could you revert reformating please ?

File: src/app/specialties/specialty-add/specialty-add.component.html [hunk 5]
```diff
@@ -19,26 +19,24 @@
...
-        </div>
+        <div *ngIf="specialityName.errors?.required">Name is required.</div>
       </div>
       <div class="form-group has-feedback">
-        <div class="form-group ">
+        <div class="form-group">
           <label class="col-sm-1 control-label">Name</label>
           <div class="col-sm-6">
-            <input id="name" name="name" class="form-control" type="text" [(ngModel)]="specialty.name" #specialityName="ngModel"/>
+            <input id="name" class="form-control" type="text" [(ngModel)]="speciality.name" name="specialityName" #specialityName="ngModel" />
```
Comment (arey) [inline src/app/specialties/specialty-add/specialty-add.component.html:34]: Why does we have to 2 `<input>`with the same `id`?

File: src/app/owners/owner-add/owner-add.component.ts [hunk 6]
```diff
@@ -32,6 +33,7 @@ import {Router} from '@angular/router';
 })
 export class OwnerAddComponent implements OnInit {
 
+  @ViewChild("ownerForm", { static: true }) ownerForm: NgForm;
```
Comment (arey) [inline src/app/owners/owner-add/owner-add.component.ts:36]: Why did you introduced the `@ViewChild`decorator?

File: src/app/owners/owner-add/owner-add.component.spec.ts [hunk 7]
```diff
@@ -22,49 +22,103 @@
...
 
 class OwnserServiceStub {
-
   addOwner(owner: Owner): Observable<Owner> {
     return of(owner);
   }
 }
 
-describe('OwnerAddComponent', () => {
+describe("OwnerAddComponent", () => {
```
Comment (arey) [inline src/app/owners/owner-add/owner-add.component.spec.ts:46]: Linting should be apply

File: karma.conf.js [hunk 8]
```diff
@@ -13,7 +13,10 @@ module.exports = function (config) {
       require('@angular-devkit/build-angular/plugins/karma')
     ],
     client:{
-      clearContext: false // leave Jasmine Spec Runner output visible in browser
+      clearContext: false, // leave Jasmine Spec Runner output visible in browser
+      jasmine:{
+        random: false
```
Comment (arey) [inline karma.conf.js:18]: Why disabling random?

File: src/app/owners/owner-add/owner-add.component.ts (hunk 6 above)
Comment (simrin051) [reply src/app/owners/owner-add/owner-add.component.ts:36]: it was throwing error as ownerForm not found in component ts file

File: karma.conf.js (hunk 8 above)
Comment (simrin051) [reply karma.conf.js:18]: To run the test cases in order without random seed

Comment (arey) [reply karma.conf.js]: Did you fall on errors due to random execution?

**Comment by simrin051** [comment]
@arey 

I have committed with changes that you have mentioned, i have also commented on few changes. May i know why a Angular CI check is failing, i do not have idea on that. Can you please advice.

File: karma.conf.js (hunk 8 above)
Comment (simrin051) [reply karma.conf.js]: No i did not..i have removed now in latest commit.

**Comment by simrin051** [comment]
Hi @arey 

tslint is not longer supporting, eslint does support as universal to typescript and js users.

**Comment by simrin051** [comment]
Hi @arey

tslint is not longer supporting, eslint does support as universal to typescript and js users. Can you please advice.

**Pull request closed without merging**